samples/subsys/zbus/benchmark/src/msg_sub.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "messages.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/zbus/zbus.h>
+
+#define CONSUMER_STACK_SIZE (CONFIG_IDLE_STACK_SIZE + CONFIG_BM_MESSAGE_SIZE)
+
+extern atomic_t count;
+
+ZBUS_CHAN_DECLARE(bm_channel);
+
+#define SFY(i, _) s##i
+
+#define CONSUMERS_NAME_LIST LISTIFY(CONFIG_BM_ONE_TO, SFY, (, /* separator */))
+
+#define CREATE_OBSERVER(s) ZBUS_MSG_SUBSCRIBER_DEFINE(s)
+
+FOR_EACH(CREATE_OBSERVER, (;), CONSUMERS_NAME_LIST)
+	;
+
+#define CREATE_OBSERVATIONS(s) ZBUS_CHAN_ADD_OBS(bm_channel, s, 3)
+
+FOR_EACH(CREATE_OBSERVATIONS, (;), CONSUMERS_NAME_LIST)
+	;
+
+#define S_TASK(name)                                                                               \
+	void name##_task(void)                                                                     \
+	{                                                                                          \
+		const struct zbus_channel *chan;                                                   \
+		struct bm_msg msg_received;                                                        \
+                                                                                                   \
+		while (1) {                                                                        \
+			if (zbus_sub_wait_msg(&name, &chan, &msg_received, K_FOREVER) == 0) {      \
+				atomic_add(&count, *((uint16_t *)msg_received.bytes));             \
+			} else {                                                                   \
+				k_oops();                                                          \
+			}                                                                          \
+		}                                                                                  \
+	}                                                                                          \
```

Comment:
Do you really have to generate these? Could you not  use the same code and `name` as argument?